tests/test_builders/xpath_util.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+from xml.etree.ElementTree import tostring
+
+if TYPE_CHECKING:
+    import os
+    from collections.abc import Callable, Sequence
+    from xml.etree.ElementTree import Element
+
+
+def _gettext(node: Element) -> str:
+    if node.text is not None:
+        # the node has only one text
+        return node.text
+
+    # the node has tags and text; gather texts just under the node
+    return ''.join(n.tail or '' for n in node)
+
+
+def check_xpath(
+    root: Element,
+    filename: str | os.PathLike[str],
+    xpath: str,
+    check: str | re.Pattern[str] | Callable[[Sequence[Element]], None] | None,
+    be_found: bool = True,
+) -> None:
+    nodes = root.findall(xpath)
```

Comment:
`mypy` reported success when I changed this to `nodes = iter(root.findall(xpath))` -- and this module isn't in the `mypy` overrides list.

Perhaps we should leave the [`list(...)` call](https://github.com/sphinx-doc/sphinx/blob/66fa790b3a41b97ccbdd61e072cb2098886a7a74/tests/test_builders/test_build_html.py#L17) in place for this one?

```suggestion
    nodes = list(root.findall(xpath))
```

Reply:
I'm adding an assert after (to ensure that we are using the correct implementation). 
I'm currently refactoring so wait a bit (I'm making a normalized API).